### PR TITLE
Update Alibaba domains in MDFP (alibabacloud.com)

### PR DESCRIPTION
**Pull request by Gowee**
As is in the title.

Domains are pulled from the Sub Alt Name section of the TLS certificate of alibabacloud.com.

Currently, I cannot find the Badger breaks any pages of alibabacloud.com or its related/affiliated websites. But I think it is necessary.

BTW, since it seems that alicdn.com is a CDN only used by Alibaba's websites instead of a public CDN. I suppose it would be better to remove it from the yellow list. But I am not sure whether this would unexpectedly break other websites or not. Need I open a separate issue discussing this?

### CODE REVIEW

**Comment by ghostwords** [comment]
>...it seems that alicdn.com is a CDN only used by Alibaba's websites instead of a public CDN. I suppose it would be better to remove it from the yellow list.

Yes, we should remove `alicdn.com`. It was added to work around breakages on Taobao, another Alibaba property (#630), before we had the MDFP system. This could be done as part of #1593 though.

**Comment by ghostwords** [comment]
Tests are failing because `net.cn` is a Public Suffix List domain, which means Privacy Badger will never match it as this time. We could add `somesubdomain.net.cn`, but not `net.cn`. It's kind of like `example.co.uk` vs. `co.uk`.

**Comment by ghostwords** [comment]
>Currently, I cannot find the Badger breaks any pages of alibabacloud.com or its related/affiliated websites. But I think it is necessary.

Could you explain why this PR is necessary then?

File: src/js/multiDomainFirstParties.js [hunk 1]
```diff
@@ -126,7 +148,6 @@ var multiDomainFirstPartiesArray = [
     "aliloan.com",
     "alipay-cloud.com",
     "alipay.cn",
-    "alipay.com",
```
Comment (ghostwords) [inline src/js/multiDomainFirstParties.js]: Does `alipay.com` not belong to Alipay?

**Comment by Gowee** [comment]
> Tests are failing because net.cn is a Public Suffix List domain
Sorry. I think it might be due to my carelessly searching-and-replacing. I will push a new commit to fix.
> Could you explain why this PR is necessary then?
If alicdn.com is removed from the yellow list, alibabacloud.com and some of its related websites might be broken AFAICS.

File: src/js/multiDomainFirstParties.js (hunk 1 above)
Comment (Gowee) [reply src/js/multiDomainFirstParties.js]: No. The domain is already present in L124(prev)/146(now). I think there should not be duplicated domains on the list? BTW, the check script should also cover this (dups)?

Comment (ghostwords) [reply src/js/multiDomainFirstParties.js]: Ah, I see, a copy-paste typo in my part. Thanks!

**Comment by ghostwords** [comment]
I think the main reason to add to MDFP only as needed and not worry about completeness is that it takes us time to compile and review these domain lists, time that we could instead be spending fixing sites that are actively broken now.

**Comment by Gowee** [comment]
I think it would be better to delay this PR at least until alicdn.com is removed from the yellow list.

File: src/js/multiDomainFirstParties.js (hunk 1 above)
Comment (ghostwords) [reply src/js/multiDomainFirstParties.js]: I added a check to prevent duplicates in the future: 6d4dbcfa8417b082277e3b9c7b88f67f97807d7e

**Comment by Gowee** [comment]
As alicdn.com is removed from the yellow list, alibabacloud.com is now broken. The time to reopen this commit? There seems no conflict with the current master.

> This list looks good, but is missing a couple of entries from the certificate. Shall we add them as well? ...
> `+    "pcs-gw-aliyun.com",`...
> `+    "r-app-aliyun.com",`...
These entries are not present in the current certificate? Do we still need these?

**Comment by Gowee** [comment]
I just found out that there are still bunches of domains belonging to Alibaba in the certificate of amap.com. And some of their pages are broken or partially broken. For example,

`https://www.alifanyi.com/`, blank homepage, due to `alicdn.com`
`https://ditu.amap.com/`, icon fonts missing, due to `alicdn.com`
`https://csupplier.alibabacorp.com/supplier/pub/index.htm`, styles broken, due to `alicdn.com`

Is it better to push new commits along with this PR or new PR for discussion?

**Comment by Gowee** [comment]
> I think the main reason to add to MDFP only as needed and not worry about completeness is that it takes us time to compile and review these domain lists, time that we could instead be spending fixing sites that are actively broken now.

Sounds reasonable. There are too many Alibaba domains to consider. Do we really need to cover them all? Are there any statistics built in the Badger so that we can check for the popularity of candidate domains for a reference to judge whether a domain is "needed" or not?

**Comment by ghostwords** [comment]
Hi @Gowee, sorry for losing track of this PR. A couple of updates:

- I just restored `alicdn.com` to the yellowlist (#2601) as it appears to be used by all sorts of non-Alibaba websites
- I added a couple of domains to Alibaba's MDFP (d70e0c3826213d795952d7300dd9083abdd98169), which introduced a conflict with this PR

We don't need to add all Alibaba domains. I can check the Privacy Badger error reports database for anything we're missing.

Could you resolve the conflict, and also add the `alifanyi.com`, `amap.com` and `alibabacorp.com` domains?

**Comment by Gowee** [comment]
> sorry for losing track of this PR. A couple of updates:

No problem.
> We don't need to add all Alibaba domains. I can check the Privacy Badger error reports database for anything we're missing.

Yes. You have already explained the reason to me. So it is still a good idea to proceed with this PR?

**Pull request merged**